sway-core/src/semantic_analysis/ast_node/expression/typed_expression/enum_instantiation.rs

```diff
@@ -35,6 +35,19 @@ pub(crate) fn instantiate_enum(
         errors
     );
 
+    if type_engine
```

Comment:
Would you add a line comment above this indicating what this block of code is doing?

Reply:
Added a line comment.